packages/@react-aria/i18n/docs/useLocale.mdx

```diff
@@ -0,0 +1,62 @@
+<!-- Copyright 2020 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License. -->
+
+import {Layout} from '@react-spectrum/docs';
+export default Layout;
+
+import docs from 'docs:@react-aria/i18n';
+import {ClassAPI, HeaderInfo, TypeContext, FunctionAPI, TypeLink} from '@react-spectrum/docs';
+import packageData from '../package.json';
+
+---
+category: Internationalization
+---
+
+# useLocale
+
+<p>{docs.exports.useLocale.description}</p>
+
+<HeaderInfo
+  packageData={packageData}
+  componentNames={['useLocale']} />
+
+## Introduction
+
+`useLocale` allows components to access the current locale and interface layout direction.
+By default, this is automatically detected based on the browser or system language, but it can
+be overridden by using the [I18nProvider](I18nProvider.html) at the root of your app.
+
+`useLocale` should be used in the root of your app to define the 
+[lang](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/lang)
+and [dir](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes/dir) attributes 
+so that the browser knows which language and direction the user interface should be rendered in.
```

Comment:
I might be wrong here/misunderstanding how this works, but does useLocale need a I18nProvider to be situated above it to pick up the correct current locale? 
useLocale basically looks for the nearest I18nContext and if it doesn't find one it defaults to  
```
locale: 'en-US',
direction: 'ltr'
```
since that is the default of the I18nContext right? But if they use I18nProvider, the default locale is looked up via useDefaultLocale and the I18nContext gets properly set to the user's current locale right?

Lemme know if I'm reading the code wrong here or if the assumption is that users should always have a I18nProvider at the top level, even if they are just using react-aria.

Reply:
mmm yeah you're right. we should probably change that so it does the detection in `useLocale` rather than the provider.